### PR TITLE
bugfix(apphealth): app health status no change and return directly.

### PR DESCRIPTION
**Pull request by 1046102779**
Signed-off-by: 1046102779 <seachen@tencent.com>

When app health status does not change, initialization or destruction operations are repeated, resulting in resource leakage

### CODE REVIEW

**Comment by 1046102779** [comment]
image

**Comment by yaron2** [comment]
Ping @1046102779

**Comment by 1046102779** [comment]
> Ping @1046102779
> 
> Is the failing unit test related to the change?

No. I verify that the unit tests are OK at locally.   I'll check it out tonight, it's weird. I only have few change.

**Comment by 1046102779** [comment]
> > Ping @1046102779
> > Is the failing unit test related to the change?
> 
> No. I verify that the unit tests are OK at locally. I'll check it out tonight, it's weird. I only have few change.

@yaron2 Help me review these PRs. The status of these PRs has been stuck for a long time.

https://github.com/dapr/go-sdk/pull/323
https://github.com/dapr/kit/pull/20
https://github.com/dapr/components-contrib/pull/2041
https://github.com/dapr/components-contrib/pull/2039
https://github.com/dapr/docs/pull/2732
https://github.com/dapr/dapr/pull/5026
https://github.com/dapr/dapr/pull/4868
https://github.com/dapr/dapr/pull/5109

**Comment by yaron2** [comment]
> > > Ping @1046102779
> > > Is the failing unit test related to the change?
> > 
> > 
> > No. I verify that the unit tests are OK at locally. I'll check it out tonight, it's weird. I only have few change.
> 
> @yaron2 Help me review these PRs. The status of these PRs has been stuck for a long time.
> 
> [dapr/go-sdk#323](https://github.com/dapr/go-sdk/pull/323) [dapr/kit#20](https://github.com/dapr/kit/pull/20) [dapr/components-contrib#2041](https://github.com/dapr/components-contrib/pull/2041) [dapr/components-contrib#2039](https://github.com/dapr/components-contrib/pull/2039) [dapr/docs#2732](https://github.com/dapr/docs/pull/2732) #5026 #4868 #5109

cc @artursouza @halspang @daixiang0

**Comment by 1046102779** [comment]
cc @yaron2  All checks have passed.

**Comment by yaron2** [comment]
@artursouza please help review this

File: tests/e2e/healthcheck/healthcheck_test.go [hunk 1]
```diff
@@ -249,41 +250,6 @@ func testAppHealthCheckProtocol(t *testing.T, protocol string) {
 			wg.Wait()
 		})
 
-		t.Run("after delay, counters aren't increasing", func(t *testing.T) {
```
Comment (artursouza) [inline tests/e2e/healthcheck/healthcheck_test.go:252]: Wait! Why is this test deleted?

**Review by artursouza — CHANGES_REQUESTED** [review]

File: tests/e2e/healthcheck/healthcheck_test.go (hunk 1 above)
Comment (yaron2) [reply tests/e2e/healthcheck/healthcheck_test.go:252]: Ping @1046102779

Comment (yaron2) [reply tests/e2e/healthcheck/healthcheck_test.go:252]: @1046102779 any reason why this test is removed?

**Comment by daixiang0** [comment]
@1046102779 please sync with main branch.

**Comment by 1046102779** [comment]
> @1046102779 please sync with main branch.

Done.

File: tests/e2e/healthcheck/healthcheck_test.go (hunk 1 above)
Comment (daixiang0) [reply tests/e2e/healthcheck/healthcheck_test.go:252]: @1046102779 ping

**Comment by daixiang0** [comment]
@1046102779 please notice the comment and reply.

**Comment by 1046102779** [comment]
This PR is closed and  create a new PR  [dapr/dapr#5412](https://github.com/dapr/dapr/pull/5412). 

the fork dapr repo was dirty.

**Pull request closed without merging**